### ISSUE TITLE
SCIM Client - a group's Keycloak ID is sent as SCIM `id` attribute instead of `externalId`

### ISSUE THREAD

**Issue by minhtule** [Captain-P-Goldfish/scim-for-keycloak#132]
When using SCIM Client, SCIM for Keycloak sends the group's KeyCloak ID as SCIM `id` attribute instead of `externalId`. For example, here is the payload for the SCIM POST request creating a new group.
```json
{
  "schemas": [
    "urn:ietf:params:scim:schemas:core:2.0:Group"
  ],
  "id": "48ac54b1-d582-493e-95d1-4c077dd57e9c",
  "displayName": "Engineering"
}
```
According to [RFC 7643 section 3.1](https://datatracker.ietf.org/doc/html/rfc7643#section-3.1), I would expect the payload to be
```json
{
  "schemas": [
    "urn:ietf:params:scim:schemas:core:2.0:Group"
  ],
  "externalId": "48ac54b1-d582-493e-95d1-4c077dd57e9c",
  "displayName": "Engineering"
}
```

This issue seems to be similar to https://github.com/Captain-P-Goldfish/scim-for-keycloak/issues/127.

**Comment by Captain-P-Goldfish** [comment]
which verson are you using? 
this was fixed with version 2.4.0.

**Comment by minhtule** [comment]
I was using 2.4.0 to test #127 which was fixed but I found this issue with Group sync.

**Comment by Captain-P-Goldfish** [comment]
fixed in 2.4.1
